framework/data/BaseDataProvider.php

```diff
@@ -44,6 +50,20 @@
     private $_keys;
     private $_models;
     private $_totalCount;
+    
+    /**
+     * @inheritdoc
+     */
+    public function init()
+    {
+        parent::init();
+        if (!isset($this->id)) {
+            if (self::$counter > 0) {
+                $this->id = 'dp-' . self::$counter;
```

Comment:
Agree with @SilverFire. Maybe create separate method that just returns this string? Something like:

```php
protected static function generateId()
{
    return 'dp-' . self::$counter;
}
```